### ISSUE TITLE
Story: Add External Link Icon to CTA Strip

### ISSUE THREAD

**Issue by andyvanavery31** [NCIOCPL/ncids#1932]
### Specifications

**Visual**

- Change the > to the external link icon
- The icon will always be fixed on the right
- External Link icon will be center aligned vertically with the button text

**Functional**
- If the link text does not have `.gov` it will get an external link icon
- A CTA Strip button can be overridden to have an external link icon
- A CTA Strip button with an external link icon can be overridden to remove it

**Impacted Components**
- [ ] NCIDS CTA Strip
  - [ ] Documentation Site update [CTA Strip](https://designsystem-dev.cancer.gov/develop/components/cta-strip)
  - [ ] Documentation Site update [Hero with CTA Strip](https://designsystem-dev.cancer.gov/develop/components/hero)
  - [ ] Storybook for [CTA Strip](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-cta-strip-default--default-long). The new story will be added as a test case.

### DoR Tasks

- [x] Comps Created
- [x] Comps Reviewed using [Checklist](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Product-Review-of-Design-Solutions)
  - [x] Developer Review
  - [x] IA Review
  - [x] Product Review
- [x] Comps Approved
  - [x] Developer Approval
  - [x] IA Approval
  - [x] Product Approval
- [x] Scenarios Created
- [x] Scenarios Reviewed
  - [x] Designer Review
  - [x] Developer Review
  - [x] IA Review
- [x] Scenarios Approved
  - [x] Designer Approval
  - [x] Developer Approval
  - [x] IA Approval
- [x] Acceptance Criteria Drafted
- [x] Acceptance Criteria Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Acceptance Criteria Approved
  - [x] Developer Approval
  - [x] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client review with demo

### Technical Notes
**Scenarios/Comps**
- Use the comps here: [CTA Strip with External Link Icon](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8420-41391&node-type=instance&m=dev). **NOTE**: This does not impact the Primary CTA on the Hero text box, do not work on that component in this ticket.
- On the [Documentation Site CTA Strip page](https://designsystem-dev.cancer.gov/develop/components/cta-strip) add:
  - In the [Best Practices section](https://designsystem-dev.cancer.gov/develop/components/cta-strip#best-practices), add a subsection which will be the last subsection with the heading, "External Links", and in the subsequent paragraph explain how and when to use and add the External Link Icon. In this section, explain how the external links are mostly "auto-magical", and provide an explanation of how to use the override.
  - Another [Code Snippet example](https://designsystem-dev.cancer.gov/develop/components/cta-strip#code-snippets):
    - Make the 1st CTA Strip button have an external link that links to www.google.com, and has the button text "Link to Google". 
    - Make the 2nd CTA Strip button an external link that links to www.google.com, has the override class to not show the External Link Icon, and has the button text, "Override External Link Icon". 
    - Make the 3rd CTA Strip button an internal link that links to www.cancer.gov, has the override class to show the External Link Icon, and has the button text, "Add External Icon to Internal Link".
- On the [Documentation Site Hero page](https://designsystem-dev.cancer.gov/develop/components/hero) add:
  - In the [Best Practices section](https://designsystem-dev.cancer.gov/develop/components/hero#best-practices), add a subsection which will be the last subsection with the heading, "External Links", and in the subsequent paragraph explain how and when to use the and add the External Link Icon. In this section, explain how the external links are mostly "auto-magical", and provide an explanation of how to use the override.
  - Another Code Snippet example after the [Default Hero with Button & Text Box and CTA Strip](https://designsystem-dev.cancer.gov/develop/components/hero#default-hero-with-button--text-box-and-cta-strip):
    - Make the 1st CTA Strip button have an external link that links to www.google.com, and has the button text "Link to Google". 
    - Make the 2nd CTA Strip button an external link that links to www.google.com, has the override class to not show the External Link Icon, and has the button text, "Override External Link Icon". 
    - Make the 3rd CTA Strip button an internal link that links to www.cancer.gov, has the override class to show the External Link Icon, and has the button text, "Add External Icon to Internal Link".
- On Storybook:
  - Within the CTA Strip folder, add a story to the [Test Cases folder](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-cta-strip-test-cases--default-long) called, "Default with External Link". 
    - Make the 1st CTA Strip button have an external link that links to www.google.com, and has the button text "Link to Google". 
    - Make the 2nd CTA Strip button an external link that links to www.google.com, has the override class to not show the External Link Icon, and has the button text, "Override External Link Icon". 
    - Make the 3rd CTA Strip button an internal link that links to www.cancer.gov, has the override class to show the External Link Icon, and has the button text, "Add External Icon to Internal Link".
  - Within the Hero folder, add a story to the [Test Cases folder](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-hero-test-cases-with-cta--with-cta-long) called, "With CTA and External Link".
    - Make the 1st CTA Strip button an external link that links to www.google.com, and has the button text "Link to Google". 
    - Make the 2nd CTA Strip button an external link that links to www.google.com, has the override class to not show the External Link Icon, and has the button text, "Override External Link Icon". 
    - Make the 3rd CTA Strip button an internal link that links to www.cancer.gov, has the override class to show the External Link Icon, and has the button text, "Add External Icon to Internal Link".

### Acceptance Criteria

```gherkin
Scenario: A CTA Strip with an external link is viewed on the NCIDS
  Given the external link icon has been added to the NCIDS Documentation Site and NCIDS Storybook for the CTA Strip
    And the CTA Strip button is linked to a url that does not contain `.gov`
  When the CTA Strip button has a linked url that does not contain `.gov`
  Then the CTA Strip button will appear with the external link icon
    And the icon on the CTA Strip button will match the comp (see links above)
    And the external link icon will remain center aligned vertically on the line of text it is on
    And the external link icon will be fixed on the right
    And the external link icon will replace the '>' on the CTA Strip button
  When a user wants to override the behavior of a CTA Strip button with a link that does not contain `.gov` to not display the external link icon
  Then they will override the behavior with an override class
    And the external link icon will not be displayed
  When a user has a CTA Strip button with a link that contains `.gov`
    And they want to override the behavior to include the external link icon
  Then they will override the behavior with an override class
    And the external link icon will be displayed
```

**Comment by bennettcc** [comment]
Per standup 1/3: this is awaiting merge of PR: https://github.com/NCIOCPL/ncids/pull/1951
